production/helm/loki/scripts/upgrade-to-6.38.0.sh

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+set -e
+
+# Loki Helm Chart Upgrade Script to 6.38.0+
+# This script safely orphans existing StatefulSets before upgrading
+
+RELEASE_NAME=""
+NAMESPACE=""
+TARGET_VERSION="6.38.0"
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+usage() {
+    echo "Usage: $0 -r RELEASE_NAME [-n NAMESPACE] [-v TARGET_VERSION]"
+    echo ""
+    echo "Options:"
+    echo "  -r RELEASE_NAME    Name of the Helm release"
+    echo "  -n NAMESPACE       Kubernetes namespace (optional, defaults to current context)"
+    echo "  -v TARGET_VERSION  Target chart version (default: 6.38.0)"
+    echo "  -h                 Show this help message"
+    echo ""
+    echo "Example:"
+    echo "  $0 -r my-loki-release -n loki-namespace -v 6.38.0"
+    exit 1
+}
+
+log() {
+    echo -e "${BLUE}[INFO]${NC} $1"
+}
+
+warn() {
+    echo -e "${YELLOW}[WARN]${NC} $1"
+}
+
+error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+    exit 1
+}
+
+success() {
+    echo -e "${GREEN}[SUCCESS]${NC} $1"
+}
+
+# Parse command line arguments
+while getopts "r:n:v:h" opt; do
+    case $opt in
+        r) RELEASE_NAME="$OPTARG" ;;
+        n) NAMESPACE="$OPTARG" ;;
+        v) TARGET_VERSION="$OPTARG" ;;
+        h) usage ;;
+        \?) error "Invalid option -$OPTARG" ;;
+    esac
+done
+
+# Validate required parameters
+if [ -z "$RELEASE_NAME" ]; then
+    error "Release name is required. Use -r to specify it."
+fi
+
+# Build namespace flag for kubectl
+NAMESPACE_FLAG=""
+if [ -n "$NAMESPACE" ]; then
+    NAMESPACE_FLAG="-n $NAMESPACE"
```

Comment:
You might want to validate that the namespace exists to fail fast using something like this

```
if [ -n "$NAMESPACE" ] && ! kubectl get namespace "$NAMESPACE" &> /dev/null; then
    error "Namespace '$NAMESPACE' does not exist"
fi
```